ckanext/saml2/plugin.py

```diff
@@ -176,6 +179,36 @@ def get_came_from(relay_state):
     return came_from.encode('utf8')
 
 
+def saml2_get_user_gen(id=None, return_gen=False):
+    query = model.Session.query(UserSsoGen).filter(or_(
+                                    UserSsoGen.user_name == id,
+                                    UserSsoGen.gen == id,
+                                    UserSsoGen.id == id))
+    if return_gen:
+        query = query.first()
+        if query is not None:
+            return query.gen if query.gen is not None else None
+    else:
+        return query if query is not None else None
+
+
+def saml_delete_user_from_saml2_sso_gen_db(query):
+    if query is not None:
+        query.delete()
+        model.Session.commit()
+
+
+def saml2_user_delete(context, data_dict):
+    query = saml2_get_user_gen(data_dict['id'])
+    gen = query.first()
```

Comment:
But `saml2_get_user_gen` can return None and you'll get unhandled exception here